apps/openmw/mwrender/npcanimation.cpp

```diff
@@ -1096,37 +1109,71 @@ const std::vector<const ESM::BodyPart *>& NpcAnimation::getBodyParts(const std::
             if (!Misc::StringUtils::ciEqual(bodypart.mRace, race))
                 continue;
 
-            bool partFirstPerson = (bodypart.mId.size() >= 3)
-                    && bodypart.mId[bodypart.mId.size()-3] == '1'
-                    && bodypart.mId[bodypart.mId.size()-2] == 's'
-                    && bodypart.mId[bodypart.mId.size()-1] == 't';
-            if(partFirstPerson != (firstPerson))
```

Comment:
Good suggestion, but we will have 4 iterations instead of one. If it is OK, I will rewrite this function.